### DISCUSSION TITLE
Is the 7900 XTX capable of ECC or ever will be?

### DISCUSSION THREAD

**Discussion by rudiservo** [ROCm/ROCm#2837] [Q&A] [answered]
I am considering buying a GPU to run ROCm for some ML personal projects, ECC would be a nice to have.

**Comment by cgmb** [comment, marked as answer]
As far as I know, the AMD Radeon™ RX 7900 XTX does not have ECC support. You may want to consider the [AMD Radeon™ PRO W7800](https://www.amd.com/en/products/professional-graphics/amd-radeon-pro-w7800) or [AMD Radeon™ PRO W7900](https://www.amd.com/en/products/professional-graphics/amd-radeon-pro-w7900) as they are also Navi 31 GPUs and both have ECC support.

**Reply by rudiservo** [reply]
Yeah, honestly W7800 is out of my budget in spite of having 32GB of vram, something with 16Gb is enough and the competition has consumer grade GPU with ECC at a much lower cost.

Hopefully AMD will figure something out.

Thanks.

**Reply by briansp2020** [reply]
Which consumer-grade GPU supports ECC?

**Reply by rudiservo** [reply]
2090, 3090, 4090 and the new non consumer RTX 2000 Ada 16GB ECC at $625 MSRP.
And yes ECC not ECR!

Hope AMD comes up with something fast.

**Reply by briansp2020** [reply]
It's news to me that 4090 supports ECC. Are you sure? I'm pretty sure no NVidia consumer-grade card supports ECC. RTX 2000 Ada 16GB is not a consumer-grade card. You are paying double the price for double the VRAM & ECC. If that class of performance suits your needs, you should buy it. 

In the current market situation, you either pay the price and get NVidia or you save some money and get AMD. But when getting AMD hardware, you need to make sure that the application you need to run works with AMD software. As far as hardware is concerned, AMD is always cheaper, because they have to price it lower due to lacking software support.

**Reply by rudiservo** [reply]
The 4090 and the 3090 Ti have ECC, although what I found online is if you try and game with it it will crash, so it's more for CUDA workloads.
Also from what I could understand GDDR6X supports ECC natively. GDRR6 != GDDR6X

If it's some kind of soft ECC I really do not know.

The ADA card although not consumer, it's super cheap with ECC, certainly much more feasible for SMB and homelabs that need ECC and/or CUDA.

Well it's an arms race to AI in general and cloud gpu is extremely expensive for SMBs, running on premise is a far cheaper alternative.

I agree that the AMD cards are much more affordable, except if you need ECC for some reason or the workload is not compatible with ROCm.

For my test case right now I want to run TabbyML and some tensorflow worklodas, it's compatible with either CUDA or ROCm, but running it for long periods of time SEU might be a thing consider.

Anyway, I hope competition comes along and ROCm continues to improve and grow, I like to see innovation.

**Reply by cgmb** [reply]
The [AMD Radeon™ PRO W7700](https://www.amd.com/en/products/professional-graphics/amd-radeon-pro-w7700) also has ECC support and 16 GB of memory, but it is Navi 32 (gfx1101) and does not appear to be supported by PyTorch. I'm not really sure why it's not enabled in AI frameworks, as I believe that it has been enabled in all ROCm libraries despite not being officially supported. That might be a viable option in the future.

**Reply by rudiservo** [reply]
Well PRO cards are outside of my budget, managed to get a 7900 XTX to play with ROCm, quite surprised in fact good enough for my current needs.
Thanks!

**Reply by tolga9009** [reply]
Just wanted to share information about 3090 Ti and 4090 ECC: you can toggle ECC mode using "Nvidia Control Panel -> 3D Settings -> Change ECC state" for these consumer grade cards. I've personally tested this on my 3090 Ti. After enabling, the driver will ask you to reboot and you're greeted with 22.5G of "ECC enabled" VRAM instead of 24G. The performance penalty is about ~10%, from what I've read online. I didn't measure it, but in my use case (Stable Diffusion) there was no noticable difference in performance or stability.

In contrast to "ordinary" system memory RAM, where I can use overclocking or memory injection to actually validate the function of ECC and error reporting, I have no experience nor knowledge which tools are used to validate ECC VRAM functionality. But atleast it's advertised and seems to have visible effects after enabling.

**Reply by rudiservo** [reply]
you probably will not notice that much of a difference in SD probably because memory throughput is not the bottleneck, you will probably see it in a substantial size LLM or anything that is memory bound.
